LykkeApi2/Automapper/RecoveryAutomapperProfile.cs

```diff
@@ -0,0 +1,51 @@
+﻿using AutoMapper;
+using Core.Domain.Recovery;
+using Core.Dto.Recovery;
+using Lykke.Service.ClientAccountRecovery.Client.AutoRestClient.Models;
+using LykkeApi2.Models.Recovery;
+
+namespace LykkeApi2.Automapper
+{
+    internal class RecoveryAutomapperProfile : Profile
+    {
+        public RecoveryAutomapperProfile()
+        {
+            CreateMap<RecoveryStartRequestModel, RecoveryStartDto>()
+                .ForMember(dest => dest.Ip,
+                    opt => opt.ResolveUsing((src, dest, destMember, context) => context.Items["Ip"]))
```

Comment:
This solution is not final, I used Autofact for easier mappings between request models.
I don't like ```ResolveUsing``` and context dictionary for passing fields which are outside of mapping models.
for example ```context.Items["Ip"]```.
```cs
var startRecoveryDto = _mapper.Map<RecoveryStartRequestModel, RecoveryStartDto>(model,
    opt =>
    {
        opt.Items["Ip"] = _requestContext.GetIp();
        opt.Items["UserAgent"] = _requestContext.UserAgent;
    });
```


Direct mapping using Object initializer was used before. It looks cumbersome, but eliminates possible errors related to context dictionary key names.
Example:
```cs
var startRecoveryDto = new RecoveryStartDto
{
    Email = model.Email,
    PartnerId = model.PartnerId,
    Ip = _requestContext.GetIp(),
    UserAgent = _requestContext.UserAgent
};
```


If it's better to construct request models directly, without using Autofac, I could return back old direct mapping approach.

Reply:
Thanks for comment.